### PR TITLE
GDB-10937 - Introduce Persistence of the Selected Chat

### PR DESCRIPTION
**Pull request by boyan-tonchev**
## What
Introduce persistence for the selected chat.

## Why
To allow the page to initialize with the same chat as the last one the user viewed during their previous visit.

## How
Save the chat ID in the browser's local storage and load it during page initialization.

### CODE REVIEW

File: test-cypress/integration/ttyg/chat-list.spec.js [hunk 1]
```diff
@@ -142,6 +143,27 @@ describe.skip('TTYG chat list', () => {
         // And the chat list should not be visible
         TTYGViewSteps.getChatsPanel().should('be.hidden');
     });
+
+    it('should persist selected chat', () => {
+        TTYGStubs.stubChatsListGet();
+        TTYGStubs.stubChatGet();
+        TTYGStubs.stubAgentListGet();
+        // Given I have opened the ttyg page
+        TTYGViewSteps.visit();
```
Comment (svilenvelikov) [inline test-cypress/integration/ttyg/chat-list.spec.js:152]: Put a wait for `cy.wait('@get-chat')` here and after line 162

File: src/js/angular/ttyg/services/ttyg-storage.service.js [hunk 2]
```diff
@@ -54,8 +54,31 @@ function TTYGStorageService(localStorageAdapter, LSKeys) {
         }
     }
 
+    /**
+     * Saves the chat in the local storage.
+     * @param {ChatModel} chat
+     */
+    function saveChat(chat) {
+        const settings = getTtygSettings();
+        settings.chat.id = chat ? chat.id : undefined;
```
[the remark this answers is not in the document]
Comment (boyan-tonchev) [reply src/js/angular/ttyg/services/ttyg-storage.service.js:63]: When user click on create a new chat button, the selected chat will be undefined

File: test-cypress/integration/ttyg/chat-list.spec.js (hunk 1 above)
Comment (boyan-tonchev) [reply test-cypress/integration/ttyg/chat-list.spec.js:152]: Done

File: src/js/angular/ttyg/services/ttyg-storage.service.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (svilenvelikov) [reply src/js/angular/ttyg/services/ttyg-storage.service.js:63]: I don't get it. If the chat is undefined, what we are persisting then?

[the remark this answers is not in the document]
Comment (boyan-tonchev) [reply src/js/angular/ttyg/services/ttyg-storage.service.js:63]: Removes the ID if one already exists.

[the remark this answers is not in the document]
Comment (svilenvelikov) [reply src/js/angular/ttyg/services/ttyg-storage.service.js:63]: Put a comment about that please!

[the remark this answers is not in the document]
Comment (boyan-tonchev) [reply src/js/angular/ttyg/services/ttyg-storage.service.js:64]: Done

**Review by svilenvelikov — APPROVED** [review]

**Pull request merged**